test_rmw_implementation/test/test_subscription.cpp

```diff
@@ -377,3 +377,56 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_with_bad_args) {
+  bool taken = false;
+  test_msgs__msg__BasicTypes output_message{};
+  test_msgs__msg__BasicTypes test_message{};
+  rmw_subscription_allocation_t * null_allocation{nullptr}; // still valid allocation
+
+  rmw_ret_t ret = rmw_take(nullptr, &output_message, &taken, null_allocation);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret) << rmw_get_error_string().str;
+  EXPECT_EQ(taken, false);
+  EXPECT_EQ(output_message, test_message);
+  rmw_reset_error();
+
+  ret = rmw_take(sub, nullptr, &taken, null_allocation);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret) << rmw_get_error_string().str;
+  EXPECT_EQ(taken, false);
+  rmw_reset_error();
+
+  ret = rmw_take(sub, &output_message, nullptr, null_allocation);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret) << rmw_get_error_string().str;
+  EXPECT_EQ(output_message, test_message);
+  rmw_reset_error();
+
+  const char * implementation_identifier = sub->implementation_identifier;
+  sub->implementation_identifier = "not-an-rmw-implementation-identifier";
+  ret = rmw_take(sub, &output_message, &taken, null_allocation);
+  EXPECT_EQ(RMW_RET_INCORRECT_RMW_IMPLEMENTATION, ret) << rmw_get_error_string().str;
+  EXPECT_EQ(taken, false);
+  EXPECT_EQ(output_message, test_message);
+  rmw_reset_error();
+  sub->implementation_identifier = implementation_identifier;
+}
+
+bool operator==(const test_msgs__msg__BasicTypes & m1, const test_msgs__msg__BasicTypes & m2)
+{
+  if (m1.bool_value == m2.bool_value &&
```

Comment:
@Lobotuerk nit: 
```suggestion
  return (m1.bool_value == m2.bool_value && ...);
```
also works.

Reply:
Updated